clojure-mode.el

```diff
@@ -776,7 +776,8 @@ Called by `imenu--generic-function'."
       (let (found?
             (deftype (match-string 2))
             (start (point)))
-        (down-list)
+        (ignore-errors
```

Comment:
I guess some comment here would be useful, otherwise it's really hard to tell why this error needs to be ignored. I'm also a bit worried this might suppress actual errors and complicate the debugging.

Reply:
I looked a bit more into it why I didn't happen before and nobody noticed.
The change to down-list came only in May with this: https://git.savannah.gnu.org/cgit/emacs.git/commit/?id=0b3b295776ce723885c9997ab26d57314db2a5df

I guess a better workaround would be to wrap it in
`unless (ppss-comment-or-string-start (syntax-ppss))` instead of `ignore-errors`,
but `ppss-comment-or-string-start` is only available since Emacs 27 and clojure-mode
is marked to support Emacs 25+.

I updated the commit message with this info and added a short comment.
Happy to rephrase it or change something else.

Reply:
If `ppss-comment-or-string-start` is not complex I guess we can just copy it and inline it.